test/fixture/server.js

```diff
@@ -64,13 +67,19 @@ module.exports.start = function(options, callback){
   }
 
   //configure wsfed middleware
-  app.get('/wsfed', 
-      wsfed.auth(xtend({}, {
-        issuer:             'fixture-test',
-        getPostURL:         getPostURL,
-        cert:               credentials.cert,
-        key:                credentials.key
-      }, options)));
+  app.get('/wsfed', function(req, res, next) {
+    wsfed.auth(xtend({}, {
+      issuer:             'fixture-test',
+      getPostURL:         getPostURL,
+      cert:               credentials.cert,
+      key:                credentials.key
+    }, module.exports.options))(req, res, function(err){
```

Comment:
Instead of doing this why not pass in a `profileMapper` stub to start and have that behave differently in each test?

Reply:
With the current implementation, if we pass stubbed profile mapper, we wouldn't be able to test the behavior of picking default profile mapper `PassportProfileMapper` (existing tests basically), since [the corresponding evaluation of which mapper to chose](https://github.com/auth0/node-wsfed/pull/20/files#diff-ef1c03e32131619adffba50a09b5a513447f8d8e34d380aacee351c755114727R36) happens on server start (at the very beginning).

Another viable option is to leave the current test suite as it is, testing the default profile mapper, and have a separate test suite where we start a server with a stubbed profile mapper.

There are different options in the end, but honestly, I find the proposed approach of runtime building wsfed auth for testing the most flexible and easier to use. I saw it basically in `node-samlp` https://github.com/auth0/node-samlp/blob/master/test/fixture/server.js#L58-L70

Reply:
The current approach is fine. 👍